airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py

```diff
@@ -0,0 +1,1256 @@
+        :param optimization_objective: Optional. Objective function the Model is to be optimized towards.
+            The training task creates a Model that maximizes/minimizes the value of the objective function
+            over the validation set.
+
+            The supported optimization objectives depend on the prediction type, and in the case of
+            classification also the number of distinct values in the target column (two distinct values
+            -> binary, 3 or more distinct values -> multi class). If the field is not set, the default
+            objective function is used.
+
+            Classification (binary):
+            "maximize-au-roc" (default) - Maximize the area under the receiver operating characteristic (ROC)
+            curve.
+            "minimize-log-loss" - Minimize log loss.
+            "maximize-au-prc" - Maximize the area under the precision-recall curve.
+            "maximize-precision-at-recall" - Maximize precision for a specified recall value.
+            "maximize-recall-at-precision" - Maximize recall for a specified precision value.
+
+            Classification (multi class):
+            "minimize-log-loss" (default) - Minimize log loss.
+
+            Regression:
+            "minimize-rmse" (default) - Minimize root-mean-squared error (RMSE).
+            "minimize-mae" - Minimize mean-absolute error (MAE).
+            "minimize-rmsle" - Minimize root-mean-squared log error (RMSLE).
+        :param column_specs: Optional. Alternative to column_transformations where the keys of the dict are
+            column names and their respective values are one of AutoMLTabularTrainingJob.column_data_types.
+            When creating transformation for BigQuery Struct column, the column should be flattened using "."
+            as the delimiter. Only columns with no child should have a transformation. If an input column has
+            no transformations on it, such a column is ignored by the training, except for the targetColumn,
+            which should have no transformations defined on. Only one of column_transformations or
+            column_specs should be passed.
+        :param column_transformations: Optional. Transformations to apply to the input columns (i.e. columns
+            other than the targetColumn). Each transformation may produce multiple result values from the
+            column's value, and all are used for training. When creating transformation for BigQuery Struct
+            column, the column should be flattened using "." as the delimiter. Only columns with no child
+            should have a transformation. If an input column has no transformations on it, such a column is
+            ignored by the training, except for the targetColumn, which should have no transformations
+            defined on. Only one of column_transformations or column_specs should be passed. Consider using
+            column_specs as column_transformations will be deprecated eventually.
```

Comment:
@MaksYermak can you please clarify this deprecation eventually?
This is the first time we added this operator why do we warn about deprecation (and if there is something that user needs to be aware of why do we warn only in docstring?)

Reply:
@eladkal thank you for your comment. It is my mistake I haven´t noticed that this parameter is deprecated in the google-cloud-aiplatform library. I will create PR with deprecation warning for this parameter.